alerting/alerter-war/src/main/java/org/hawkular/metrics/alerter/groups/GroupTriggerManager.java

```diff
@@ -0,0 +1,597 @@
+
+        mgtExecutor = new ScheduledThreadPoolExecutor(THREAD_POOL_SIZE);
+
+        refresh();
+
+        if (null == definitionsListener) {
+            log.info("Registering Trigger UPDATE/REMOVE listener");
+            definitionsListener = new DefinitionsListener() {
+                @Override
+                public void onChange(Set<DefinitionsEvent> event) {
+                    if (coordinator) {
+                        log.warnf("Refreshing due to change event %s", event);
+                        refresh();
+                    }
+                }
+            };
+            definitions.registerListener(definitionsListener, DefinitionsEvent.Type.TRIGGER_CONDITION_CHANGE,
+                    DefinitionsEvent.Type.TRIGGER_REMOVE);
+        }
+    }
+
+    @PreDestroy
+    public void shutdown() {
+        if (coordinator) {
+            stop();
+        }
+
+        if (distributed) {
+            cacheManager.removeListener(topologyChangeListener);
+            cacheManager.stop();
+        }
+    }
+
+    public void stop() {
+        log.infof("Stopping Hawkular Metrics Group Trigger Manager, distributed=%s", distributed);
+
+        if (null != mgtFutures) {
+            mgtFutures.values().forEach(f -> f.cancel(true));
+        }
+        if (null != mgtExecutor) {
+            mgtExecutor.shutdown();
+            mgtExecutor = null;
+        }
+    }
+
+    public void cancel(Trigger mgt) {
+        ScheduledFuture<?> f = mgtFutures.get(mgt);
+        if (null != f) {
+            log.warnf("Canceling Hawkular Metrics Group Trigger Job for %s", mgt);
+            try {
+                f.cancel(true);
+                mgtFutures.remove(mgt);
+            } catch (Exception e) {
+                log.warnf("Failed to cancel Hawkular Metrics Group Trigger Job for %s: %s", mgt, e.getMessage());
+            }
+        }
+    }
+
+    private void refresh() {
+        try {
+            log.warn("Refreshing Metrics Group Triggers!");
+
+            // get all of the metrics group triggers (tagged for hawkular metrics)
+            Collection<Trigger> taggedMgts = definitions.getAllTriggersByTag(
+                    TAG_GROUP_TRIGGER,
+                    TAG_GROUP_TRIGGER_VALUE);
+            log.warnf("Found [%s] Metrics Group Triggers!", taggedMgts.size());
+
+            // generate valid set of metrics group trigger ids
+            Set<Trigger> mgts = taggedMgts.stream()
+                    .filter(t -> {
+                        if (!t.isGroup()) {
+                            log.warnf("Tagged as Metrics Group Trigger is not a group trigger, skipping %s", t);
+                            return false;
+                        }
+                        return true;
+                    })
+                    .collect(Collectors.toSet());
+
+            // for each metrics group trigger run a job to manage its member trigger set
+            for (Trigger mgt : mgts) {
+                if (mgtFutures.containsKey(mgt)) {
```

Comment:
>Ok, so it is the job who is periodically query the backend to synchronize the trigger and metric definitions.

That's correct.  It's a repeating job, one per MGT, and on each run it synchronizes the member triggers for that group trigger, based on the current MGT tag values and the resulting metrics from the dataId tag queries.